plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -62,15 +65,26 @@ public IcebergPageSource(
         // requiredColumns should include all expectedColumns as well as any columns needed by the DeleteFilter
         requireNonNull(expectedColumns, "expectedColumns is null");
         requireNonNull(requiredColumns, "requiredColumns is null");
+        requireNonNull(deleteFilterColumns, "deleteFilterColumns is null");
+        requireNonNull(rowPositionColumnIndex, "rowPositionColumnIndex is null");
         this.expectedColumnIndexes = new int[expectedColumns.size()];
-        for (int i = 0; i < expectedColumns.size(); i++) {
-            IcebergColumnHandle expectedColumn = expectedColumns.get(i);
-            checkArgument(expectedColumn.equals(requiredColumns.get(i)), "Expected columns must be a prefix of required columns");
-            expectedColumnIndexes[i] = i;
-
-            if (expectedColumn.isMergeRowIdColumn()) {
+        this.deleteFilterIndexes = new int[deleteFilterColumns.size()];
+        for (int i = 0; i < requiredColumns.size(); i++) {
```

Comment:
Is `expectedColumns` always subset of `requiredColumns` with index of elements preservation?